packages/mui-base/src/SnackbarUnstyled/SnackbarUnstyled.tsx

```diff
@@ -106,6 +107,14 @@ const SnackbarUnstyled = React.forwardRef(function SnackbarUnstyled(
     className: classes.root,
   });
 
+  const transitionProps: WithOptionalOwnerState<SnackbarUnstyledTransitionSlotProps> = useSlotProps(
```

Comment:
You can use `getSlotProps` and propagate the `getTransitionProps` function there, for example: https://github.com/mui/material-ui/blob/master/packages/mui-base/src/InputUnstyled/InputUnstyled.tsx#L140

Reply:
The types were misbehaving so I thought to keep it simple by spreading it to the component itself but now I figured it out.

Reply:
I wonder if we should even care about transitions in unstyled components - not sure.

Reply:
I just reevaluated it and found out that it's not needed. Do take a look at the [second demo](https://deploy-preview-33227--material-ui.netlify.app/base/react-snackbar/#transitions) related to transition and let me know what you think about the approach.